src/status_im/contexts/wallet/wallet_connect/responding_events.cljs

```diff
@@ -78,22 +86,38 @@
             [{:method     "wallet_buildTransaction"
               :params     [chain-id (.stringify js/JSON raw-data)]
               :on-success on-success
-              :on-error   [:wallet-connect/on-sign-error]}]]]})))
+              :on-error   [:wallet-connect/on-sign-error screen]}]]]})))
 
 (rf/reg-event-fx
- :wallet-connect/respond-sign-transaction-data
+ :wallet-connect/respond-send-transaction-data
  (fn [_ [password data]]
    (let [{:keys [address messageToSign]} data]
      {:fx [[:effects.wallet-connect/sign-message
             {:password   password
              :address    address
              :data       messageToSign
-             :on-error   #(rf/dispatch [:wallet-connect/on-sign-error %])
-             :on-success #(rf/dispatch [:wallet-connect/send-response %])}]]})))
+             :on-error   #(rf/dispatch [:wallet-connect/on-sign-error
+                                        :screen/wallet-connect.send-transaction %])
+             :on-success #(rf/dispatch [:wallet-connect/send-response
+                                        :screen/wallet-connect.send-transaction %])}]]})))
+
+(rf/reg-event-fx
+ :wallet-connect/respond-sign-transaction-data
+ (fn [_ [password data]]
+   (let [_ (js/console.log "ALWX DATA" (clj->js data))
+         {:keys [address messageToSign]} data]
+     {:fx [[:effects.wallet-connect/sign-message
+            {:password   password
+             :address    address
+             :data       messageToSign
+             :on-error   #(rf/dispatch [:wallet-connect/on-sign-error
+                                        :screen/wallet-connect.sign-transaction %])
+             :on-success #(rf/dispatch [:wallet-connect/send-response
+                                        :screen/wallet-connect.sign-transaction %])}]]})))
 
 (rf/reg-event-fx
  :wallet-connect/on-sign-error
- (fn [{:keys [db]} [error]]
+ (fn [{:keys [db]} [screen error]]
```

Comment:
Instead of passing the screen through all the events, can't we use the `method-to-screen` function here?